commit dccf78d39f1069a5ddf4328bf0c97aa5f2f4296e
Author: Baoquan He <bhe@redhat.com>
Date:   Tue Nov 28 13:44:57 2023 +0800

    kernel/Kconfig.kexec: drop select of KEXEC for CRASH_DUMP
    
    Ignat Korchagin complained that a potential config regression was
    introduced by commit 89cde455915f ("kexec: consolidate kexec and crash
    options into kernel/Kconfig.kexec").  Before the commit, CONFIG_CRASH_DUMP
    has no dependency on CONFIG_KEXEC.  After the commit, CRASH_DUMP selects
    KEXEC.  That enforces system to have CONFIG_KEXEC=y as long as
    CONFIG_CRASH_DUMP=Y which people may not want.
    
    In Ignat's case, he sets CONFIG_CRASH_DUMP=y, CONFIG_KEXEC_FILE=y and
    CONFIG_KEXEC=n because kexec_load interface could have security issue if
    kernel/initrd has no chance to be signed and verified.
    
    CRASH_DUMP has select of KEXEC because Eric, author of above commit, met a
    LKP report of build failure when posting patch of earlier version.  Please
    see below link to get detail of the LKP report:
    
        https://lore.kernel.org/all/3e8eecd1-a277-2cfb-690e-5de2eb7b988e@oracle.com/T/#u
    
    In fact, that LKP report is triggered because arm's <asm/kexec.h> is
    wrapped in CONFIG_KEXEC ifdeffery scope.  That is wrong.  CONFIG_KEXEC
    controls the enabling/disabling of kexec_load interface, but not kexec
    feature.  Removing the wrongly added CONFIG_KEXEC ifdeffery scope in
    <asm/kexec.h> of arm allows us to drop the select KEXEC for CRASH_DUMP.
    Meanwhile, change arch/arm/kernel/Makefile to let machine_kexec.o
    relocate_kernel.o depend on KEXEC_CORE.
    
    Link: https://lkml.kernel.org/r/20231128054457.659452-1-bhe@redhat.com
    Fixes: 89cde455915f ("kexec: consolidate kexec and crash options into kernel/Kconfig.kexec")
    Signed-off-by: Baoquan He <bhe@redhat.com>
    Reported-by: Ignat Korchagin <ignat@cloudflare.com>
    Tested-by: Ignat Korchagin <ignat@cloudflare.com>       [compile-time only]
    Tested-by: Alexander Gordeev <agordeev@linux.ibm.com>
    Reviewed-by: Eric DeVolder <eric_devolder@yahoo.com>
    Tested-by: Eric DeVolder <eric_devolder@yahoo.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>

diff --git a/kernel/Kconfig.kexec b/kernel/Kconfig.kexec
index 7aff28ded2f4..1cc3b1c595d7 100644
--- a/kernel/Kconfig.kexec
+++ b/kernel/Kconfig.kexec
@@ -97,7 +97,6 @@ config CRASH_DUMP
 	depends on ARCH_SUPPORTS_KEXEC
 	select CRASH_CORE
 	select KEXEC_CORE
-	select KEXEC
 	help
 	  Generate crash dump after being started by kexec.
 	  This should be normally only set in special crash dump kernels